tests/src/test/scala/it/unibo/scafi/test/functional/stdlib/TestFieldUtils.scala

```diff
@@ -0,0 +1,205 @@
+    // ACT
+    exec(new TestProgram {
+      override def main(): Any = (
+        excludingSelf.sumHood(1),
+        includingSelf.sumHood(1)
+      )
+    }, ntimes = fewRounds)(net)
+
+    //ASSERT
+    assertNetworkValues((0 to 8).zip(List(
+      (net.neighbourhood(0).size, net.neighbourhood(0).size + 1), (net.neighbourhood(1).size, net.neighbourhood(1).size + 1), (net.neighbourhood(2).size, net.neighbourhood(2).size + 1),
+      (net.neighbourhood(3).size, net.neighbourhood(3).size + 1), (net.neighbourhood(4).size, net.neighbourhood(4).size + 1), (net.neighbourhood(5).size, net.neighbourhood(5).size + 1),
+      (net.neighbourhood(6).size, net.neighbourhood(6).size + 1), (net.neighbourhood(7).size, net.neighbourhood(7).size + 1), (net.neighbourhood(8).size, net.neighbourhood(8).size + 1)
+    )).toMap)(net)
+  }
+
+  Field_Utils should "support sumHood - simplified" in new SimulationContextFixture {
+    // ACT
+    net.addSensor[Int]("snsSumHood", 0)
+    net.chgSensorValue("snsSumHood", Set(0,1,2), 10)
+
+    exec(new TestProgram {
+      override def main(): Any = (
+        excludingSelf.sumHood(nbr{sense[Int]("snsSumHood")}),
+        includingSelf.sumHood(nbr{sense[Int]("snsSumHood")})
+      )
+    }, ntimes = fewRounds)(net)
+
+    //ASSERT
+    assertNetworkValues((0 to 8).zip(List(
+      (10,20), (20,30), (10,20),
+      (20,20), (30,30), (20,20),
+      (0,0), (0,0), (0,0)
+    )).toMap)(net)
+
+    /*Sensor value changes*/
+
+    net.addSensor[Int]("snsSumHood", 0)
+    net.chgSensorValue("snsSumHood", Set(0,1,2), 15)
+    net.chgSensorValue("snsSumHood", Set(4), -20)
+
+    exec(new TestProgram {
+      override def main(): Any = (
+        excludingSelf.sumHood(nbr{sense[Int]("snsSumHood")}),
+        includingSelf.sumHood(nbr{sense[Int]("snsSumHood")})
+      )
+    }, ntimes = fewRounds)(net)
+
+    //ASSERT
+    assertNetworkValues((0 to 8).zip(List(
+      (-5,10), (10,25), (-5,10),
+      (10,10), (45,25), (10,10),
+      (-20,-20), (-20,-20), (0,0)
+    )).toMap)(net)
+  }
+
+  Field_Utils should "support unionHood" in new SimulationContextFixture {
+    // ACT
+    exec(new TestProgram {
+      override def main(): Any = (
+        excludingSelf.unionHood(nbr(mid())),
+        includingSelf.unionHood(nbr(mid()))
+      )
+    }, ntimes = fewRounds)(net)
+
+    //ASSERT
+    assertNetworkValues((0 to 8).zip(List(
+      (net.neighbourhood(0), net.neighbourhood(0) + 0), (net.neighbourhood(1), net.neighbourhood(1) + 1), (net.neighbourhood(2), net.neighbourhood(2) + 2),
+      (net.neighbourhood(3), net.neighbourhood(3) + 3), (net.neighbourhood(4), net.neighbourhood(4) + 4), (net.neighbourhood(5), net.neighbourhood(5) + 5),
+      (net.neighbourhood(6), net.neighbourhood(6) + 6), (net.neighbourhood(7), net.neighbourhood(7) + 7), (net.neighbourhood(8), net.neighbourhood(8) + 8)
+    )).toMap)(net)
+  }
+
+  Field_Utils should "support mergeHoodFirst" in new SimulationContextFixture {
```

Comment:
This test does not actually test any "merging" (which is the very feature of `mergeHoodFirst`). Merging happens when a neighbour provides a map with keys that are already present in the folding map (because other neighbours are providing them). This version, `mergeHoodFirst`, simply keeps the former entry, and discards the current entry. 
You could also add a test for `mergeHood` which does the opposite (keeps the current entry and discards the former one), by properly giving a corresponding `overwritePolicy`.